baselines/depthfl/README.md

```diff
@@ -0,0 +1,162 @@
+
+# activate the environment
+poetry shell
+
+# install PyTorch with GPU support. 
+pip install torch==1.13.1+cu116 torchvision==0.14.1+cu116 torchaudio==0.13.1 --extra-index-url https://download.pytorch.org/whl/cu116
+```
+
+
+## Running the Experiments
+
+To run this DepthFL, first ensure you have activated your Poetry environment (execute `poetry shell` from this directory), then:
+
+```bash
+python -m depthfl.main # this will run using the default settings in the `conf/config.yaml`
+
+# you can override settings directly from the command line
+python -m depthfl.main exclusive_learning=true model_size=1 # exclusive learning - 100% (a)
+python -m depthfl.main exclusive_learning=true model_size=4 # exclusive learning - 25% (d)
+python -m depthfl.main fit_config.feddyn=false fit_config.kd=false # DepthFL (FedAvg)
+python -m depthfl.main fit_config.feddyn=false fit_config.kd=false fit_config.extended=false # InclusiveFL
+```
+
+To run using HeteroFL:
+```bash
+# since sbn takes too long, we test global model every 50 rounds. 
+python -m fedprox.main --config-name="heterofl" # HeteroFL
+python -m fedprox.main --config-name="heterofl" exclusive_learning=true model_size=1 # exclusive learning - 100% (a)
+```
+
+
+## Expected Results
+
+With the following command we run DepthFL (FedDyn / FedAvg), InclusiveFL, and HeteroFL to replicate the results of table 2,3,4 in DepthFL paper. 
+
+```bash
+python -m depthfl.main --config-name="heterofl" 
+python -m depthfl.main --config-name="heterofl" exclusive_learning=true model_size=1 model.scale=false
+python -m depthfl.main --config-name="heterofl" exclusive_learning=true model_size=2 model.scale=false
+python -m depthfl.main --config-name="heterofl" exclusive_learning=true model_size=3 model.scale=false
+python -m depthfl.main --config-name="heterofl" exclusive_learning=true model_size=4 model.scale=false
+
+python -m depthfl.main fit_config.feddyn=false fit_config.kd=false  
+python -m depthfl.main fit_config.feddyn=false fit_config.kd=false  exclusive_learning=true model_size=1
+python -m depthfl.main fit_config.feddyn=false fit_config.kd=false  exclusive_learning=true model_size=2
+python -m depthfl.main fit_config.feddyn=false fit_config.kd=false  exclusive_learning=true model_size=3
+python -m depthfl.main fit_config.feddyn=false fit_config.kd=false  exclusive_learning=true model_size=4
+
+python -m depthfl.main 
+python -m depthfl.main exclusive_learning=true model_size=1
+python -m depthfl.main exclusive_learning=true model_size=2
+python -m depthfl.main exclusive_learning=true model_size=3
+python -m depthfl.main exclusive_learning=true model_size=4
+
+python -m depthfl.main fit_config.feddyn=false fit_config.kd=false fit_config.extended=false
+
+python -m depthfl.main fit_config.kd=false
+```
+
+The above commands would generate results in DepthFL paper. The numbers below are the results of a single run, and although they do not perfectly match the numbers recorded in the paper, they are very close.
```

Comment:
Could you indicate in the commands above which table of results each generate? Maybe you can separate those command above into different code blocks (one Table 2, another Tab3 and another for Tab4)